Remove all utility classes, redesigned `.filter_form`

jrubenoff opened the pull request:
Remove all utility classes from `shame.scss`.

This task involved creating a more streamlined look for `.filter_form`:

![screen shot 2016-02-18 at 6 01 18 pm](https://cloud.githubusercontent.com/assets/1328849/13164389/a68f282a-d669-11e5-8da6-0db86577112e.png)

There are also a few minor styling fixes and additions, like left-aligning dropdown text within `.sidebar_sub_actions`.)

ajb commented on vendor/assets/stylesheets/dvl/components/blank_slate.scss:
```diff
@@ -35,6 +35,11 @@
   }
 }
 
+.blank_slate_alternative {
```
How should this be used? We should put it in the docs...

ajb commented on vendor/assets/stylesheets/dvl/components/sidebar_sub_actions.scss:
```diff
@@ -5,5 +5,8 @@
   font-size: $fontSmall;
   .sidebar_box + & {
```
I know this isn't part of your commit, but I think this is a hack.... if we want to conditionally display the sidebar sub actions on the right vs. the left, we should do that with a modifier, e.g. `sidebar_sub_actions.is_right`, yeah?

jrubenoff replied:
We always align `.sidebar_box` to the right and `.sidebar_nav` to the left, so the styles here match the intent of the design. Not a hack.

ajb commented on vendor/assets/stylesheets/dvl/components/sidebar_sub_actions.scss:
```diff
@@ -5,5 +5,8 @@
   .sidebar_box + & {
     text-align: right;
+    .dropdown_body {
+      text-align: left;
```
Let's add a dropdown to the docs so we can see this without running a separate app?

ajb commented:
Nice, glad that we're finally getting this done.

Added some comments inline. Also:

![img](http://take.ms/MghWqZ)

![img](http://take.ms/61sE2)

ajb commented on vendor/assets/stylesheets/dvl/components/filter_form.scss:
```diff
@@ -1,81 +1,60 @@
-      display: inline-block;
-      vertical-align: top;
+  // Search button is present but hidden
+  &.full {
```
Are we using this modifier in our apps? If no, let's remove. If yes, let's add to docs.

ajb replied:
It looks like it might have been renamed to `.without_button`?

jrubenoff commented:
I don't think we need to add a max-width to `.filter_form`. We might want full- or half-width search bars, for instance, depending on the needs of the app.

Addressed all of your comments, save for the `.sidebar_box` thing.

ajb commented:
Not sure where to put this comment, but I'm not totally satisfied with the new filter form design when there are multiple inputs, like in https://github.com/dobtco/platform/pull/497.

Here's another example from Autodrip:

![img](http://take.ms/pxTIt)

What do you think of redesigning https://github.com/dobtco/platform/pull/497 so all the inputs fit on one line?

jrubenoff commented:
It's not really the same component. My main priority there was to avoid writing a ton of `.filter_form` style overrides for something users will never see. FWIW, I like how it visually separates the filters from the primary input. Helpful on the DOBT admin orgs page, to distinguish the selectize from the selects.

But I also don't really care. We could convert all of them to `.dropdown_rows` if you want:

![screen shot 2016-02-19 at 10 22 02 am](https://cloud.githubusercontent.com/assets/1328849/13184797/c1d865c2-d6f2-11e5-8ed7-2e34778294b0.png)

ajb commented:
What do you mean by `dropdown_rows`?

ajb commented:
Ah, nm, I see in https://github.com/dobtco/platform/pull/497/files

ajb commented:
:+1: :+1: :+1:

ajb commented:
We're redesigning the "Your plan" page so this doesn't really matter, but here's what this change did to it:

![img](http://take.ms/ObPHSi)

How can we be sure this bug wasn't triggered elsewhere?

jrubenoff commented:
This is desired behavior... the input should take up the full content area unless we tell it otherwise.

It looks a little weird on the Your plan page, but that's because that page's layout is rough in the first place. Hence dobtco/screendoor-v2#3289
